pdns/recursordist/test-recursorcache_cc.cc

```diff
@@ -906,4 +906,174 @@ BOOST_AUTO_TEST_CASE(test_RecursorCache_Wipe)
   BOOST_CHECK_EQUAL(MRC.ecsIndexSize(), 0U);
 }
 
+BOOST_AUTO_TEST_CASE(test_RecursorCacheTagged)
+{
+  MemRecursorCache MRC;
+
+  std::vector<DNSRecord> records;
+  std::vector<std::shared_ptr<DNSRecord>> authRecords;
+  std::vector<std::shared_ptr<RRSIGRecordContent>> signatures;
+  time_t now = time(nullptr);
+
+  BOOST_CHECK_EQUAL(MRC.size(), 0U);
+  // An entry withoug edns subnet gets stored without tag as well
+  MRC.replace(now, DNSName("hello"), QType(QType::A), records, signatures, authRecords, true, boost::none, boost::none);
+  MRC.replace(now, DNSName("hello"), QType(QType::A), records, signatures, authRecords, true, boost::none, string("mytag"));
+  BOOST_CHECK_EQUAL(MRC.size(), 1U);
+  BOOST_CHECK_EQUAL(MRC.doWipeCache(DNSName("hello"), false, QType::A), 1U);
+  BOOST_CHECK_EQUAL(MRC.size(), 0U);
+  BOOST_CHECK_EQUAL(MRC.bytes(), 0U);
+
+  uint64_t counter = 0;
+  try {
+    for (counter = 0; counter < 100; ++counter) {
+      DNSName a = DNSName("hello ") + DNSName(std::to_string(counter));
+      BOOST_CHECK_EQUAL(DNSName(a.toString()), a);
+
+      MRC.replace(now, a, QType(QType::A), records, signatures, authRecords, true, boost::none, string("mytagA"));
+      if (!MRC.doWipeCache(a, false))
+        BOOST_FAIL("Could not remove entry we just added to the cache!");
+      MRC.replace(now, a, QType(QType::A), records, signatures, authRecords, true, boost::none, string("mytagB"));
+    }
+
+    BOOST_CHECK_EQUAL(MRC.size(), counter);
+
+    uint64_t delcounter = 0;
+    for (delcounter = 0; delcounter < counter / 100; ++delcounter) {
+      DNSName a = DNSName("hello ") + DNSName(std::to_string(delcounter));
+      BOOST_CHECK_EQUAL(MRC.doWipeCache(a, false, QType::A), 1U);
+    }
+
+    BOOST_CHECK_EQUAL(MRC.size(), counter - delcounter);
+
+    std::vector<DNSRecord> retrieved;
+    ComboAddress who("192.0.2.1");
+    int64_t matches = 0;
+    int64_t expected = counter - delcounter;
+    int64_t delstart = delcounter;
+
+    for (; delcounter < counter; ++delcounter) {
+      if (MRC.get(now, DNSName("hello ") + DNSName(std::to_string(delcounter)), QType(QType::A), false, &retrieved, who, boost::none)) {
```

Comment:
```suggestion
      if (MRC.get(now, DNSName("hello ") + DNSName(std::to_string(delcounter)), QType(QType::A), false, &retrieved, who, boost::none) > 0) {
```

Reply:
Actually this makes the test fail. This is an existing because the entries are inserted with a ttd of 0 in this case. So get() returns -1. Also an issue with the original test I copied from.

Reply:
The ttd 0 happens because records is actually empty.